// werift-rtp/lib/rtcp/header.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RtcpHeader = exports.HEADER_SIZE = void 0;
const helper_1 = require("../helper");
const utils_1 = require("../utils");
exports.HEADER_SIZE = 4;
class RtcpHeader {
    constructor(props = {}) {
        this.version = 0;
        this.padding = false;
        this.count = 0;
        this.type = 0;
        this.length = 0;
        Object.assign(this, props);
    }
    serialize() {
        const v_p_rc = { ref: 0 };
        utils_1.setBit(v_p_rc, 2, 1);
        if (this.padding)
            utils_1.setBit(v_p_rc, 1, 2);
        utils_1.setBit(v_p_rc, this.count, 3, 5);
        const buf = helper_1.bufferWriter([1, 1, 2], [v_p_rc.ref, this.type, this.length]);
        return buf;
    }
    static deSerialize(buf) {
        const [v_p_rc, type, length] = helper_1.bufferReader(buf, [1, 1, 2]);
        const version = utils_1.getBit(v_p_rc, 0, 2);
        const padding = utils_1.getBit(v_p_rc, 2, 1) > 0;
        const count = utils_1.getBit(v_p_rc, 3, 5);
        return new RtcpHeader({ version, padding, count, type, length });
    }
}
exports.RtcpHeader = RtcpHeader;
/*
 *  0                   1                   2                   3
 *  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 * |V=2|P|    RC   |      PT       |             length            |
 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 */
